### ISSUE TITLE
[clang-tidy] performance-unnecessary-value-param fix adds `#include <utility>` before the existing headers

### ISSUE THREAD

**Issue by joker-eph** [llvm/llvm-project#52859]
<img width="646" alt="Screen Shot 2021-12-23 at 1 34 44 PM" src="https://user-images.githubusercontent.com/3372300/147293298-1c60af4d-ea39-4d82-92fc-f0f6fae5fe32.png">

This is breaking the LLVM convention of adding such headers after the existing ones.

**Comment by firewave** [comment]
Looks like this might along the lines of #58221 i.e. clang-tidy is not responsible for formatting code.

**Comment by vbvictor** [comment]
This should be done by https://clang.llvm.org/extra/clang-tidy/checks/llvm/include-order.html and clang-format.

**Comment by joker-eph** [comment]
I don't believe so: when the headers are grouped (separated by an empty line) the tools are meant to handle the groups in isolation.

Minimal example:

```c++
#include <utility>

#include "mlir/IR/Operation.h"
```

Neither clang-tidy nor clang-format are acting here.

If I remove the empty line, clang-format will reorder, and clang-tidy start to warn:

```
warning: #includes are not sorted properly [llvm-include-order]
    1 | #include <utility>
      | ^        ~~~~~~~~~
      |          "mlir/IR/Operation.h"
    2 | #include "mlir/IR/Operation.h"
      |          ~~~~~~~~~~~~~~~~~~~~~
      |          <utility>
```

**Comment by vbvictor** [comment]
Then to me this is FN for llvm-include-order, it should handle system includes across groups correctly. 
It should be fairly easy: all includes that are `<>` are system thus should go in "latest group"

**Comment by joker-eph** [comment]
That seems like breaking the convention that the user can control with "groups" of headers, which the tools should respect.

Why are we adding this header with an empty line separating from the other includes here?

**Comment by vbvictor** [comment]
> Why are we adding this header with an empty line separating from the other includes here?

Sure, this can be done as well, didn't think about it initially.
I suppose this should be solution to whole problem
